src/shared/components/oncoprint/OncoprintUtils.ts

```diff
@@ -804,6 +802,95 @@ export function makeGeneticTrackWith({
+                                    string,
+                                    any[]
+                                > = {};
+                                let sequencedSampleKeysForGroup: Record<
+                                    string,
+                                    any[]
+                                > = {};
+                                if (sampleMode) {
+                                    sequencedSampleKeysForGroup = _(
+                                        geneSymbolArray
+                                    )
+                                        .keyBy(gene => gene)
+                                        .mapValues(gene => {
+                                            // you want to return only the patient ids which are listed as
+                                            // sequenced for the gene(s) in this track (plural in case of merged tracks)
+                                            return _.intersection(
+                                                _.flatMap(
+                                                    geneSymbolArray,
+                                                    gene =>
+                                                        sequencedSampleKeysByGene[
+                                                            gene
+                                                        ]
+                                                ),
+                                                groupData.map((d: any) => d.uid)
+                                            );
+                                        })
+                                        .value();
+                                } else {
+                                    sequencedPatientKeysForGroup = _(
+                                        geneSymbolArray
+                                    )
+                                        .keyBy(gene => gene)
+                                        .mapValues(gene => {
+                                            // you want to return only the patient ids which are listed as
+                                            // sequenced for the gene(s) in this track (plural in case of merged tracks)
+                                            return _.intersection(
+                                                _.flatMap(
+                                                    geneSymbolArray,
+                                                    gene =>
+                                                        sequencedPatientKeysByGene[
+                                                            gene
+                                                        ]
+                                                ),
+                                                groupData.map((d: any) => d.uid)
+                                            );
+                                        })
+                                        .value();
```

Comment:
this `else` block is almost identical to the `if` block above except the variables `sequencedSampleKeysByGene` vs `sequencedPatientKeysByGene`. probably better to extract the logic to a function and reuse.